Arduino build (verify) fails due to unrecognized types in variant.h

## Problem 

When building the library using Arduino-ide, or VS code Arduino extension, the build fails because types defined in variant.h are not recognized by oher files.

## Steps

1. Include the lbirary
2. write some code that uses it
3. build

### Error Log

Output from **VS Code Arduino extension community edition**:

```text

[Starting] Verifying sketch 'gurux_arduino.ino'
Please see the build logs in output path: ~/XLabs/Projects/gurux_arduino/build

In file included from ~/Arduino/libraries/gx_dlms_c/src/include/objectarray.h:41,
                 from ~/Arduino/libraries/gx_dlms_c/src/include/dlmssettings.h:42,
                 from ~/Arduino/libraries/gx_dlms_c/src/include/apdu.h:36,
                 from ~/Arduino/libraries/gx_dlms_c/src/gx_dlms_c.h:12,
                 from ~/XLabs/Projects/gurux_arduino/GXDLMSClient.h:39,
                 from ~/XLabs/Projects/gurux_arduino/GXDLMSClient.cpp:35:
~/Arduino/libraries/gx_dlms_c/src/include/gxobjects.h:430:9: error: 'dlmsVARIANT' does not name a type
  430 |         dlmsVARIANT value;
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/gxobjects.h:444:9: error: 'dlmsVARIANT' does not name a type
  444 |         dlmsVARIANT value;
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/gxobjects.h:3426:9: error: 'bitArray' does not name a type
 3426 |         bitArray toneMap;
      |         ^~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/gxobjects.h:3438:9: error: 'bitArray' does not name a type
 3438 |         bitArray txCoeff;
      |         ^~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/gxobjects.h:3484:9: error: 'bitArray' does not name a type
 3484 |         bitArray toneMask;
      |         ^~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/gxobjects.h:3715:9: error: 'variantArray' does not name a type
 3715 |         variantArray groupTable;
      |         ^~~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/gxobjects.h:3765:9: error: 'variantArray' does not name a type
 3765 |         variantArray destinationAddress;
      |         ^~~~~~~~~~~~
In file included from ~/Arduino/libraries/gx_dlms_c/src/include/dlmssettings.h:44:
~/Arduino/libraries/gx_dlms_c/src/include/gxvalueeventargs.h:48:9: error: 'dlmsVARIANT' does not name a type
   48 |         dlmsVARIANT value;
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/gxvalueeventargs.h:79:9: error: 'dlmsVARIANT' does not name a type
   79 |         dlmsVARIANT parameters;
      |         ^~~~~~~~~~~
In file included from ~/Arduino/libraries/gx_dlms_c/src/include/dlmssettings.h:45:
~/Arduino/libraries/gx_dlms_c/src/include/replydata.h:72:9: error: 'dlmsVARIANT' does not name a type
   72 |         dlmsVARIANT dataValue;
      |         ^~~~~~~~~~~
In file included from ~/Arduino/libraries/gx_dlms_c/src/include/client.h:46,
                 from ~/Arduino/libraries/gx_dlms_c/src/gx_dlms_c.h:19:
~/Arduino/libraries/gx_dlms_c/src/include/dlms.h:104:9: error: 'dlmsVARIANT' has not been declared
  104 |         dlmsVARIANT* value);
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/dlms.h:119:9: error: 'dlmsVARIANT' has not been declared
  119 |         dlmsVARIANT* value);
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/dlms.h:143:9: error: 'dlmsVARIANT' has not been declared
  143 |         dlmsVARIANT* value);
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/dlms.h:183:9: error: 'dlmsVARIANT' was not declared in this scope
  183 |         dlmsVARIANT* value,
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/dlms.h:183:22: error: 'value' was not declared in this scope
  183 |         dlmsVARIANT* value,
      |                      ^~~~~
~/Arduino/libraries/gx_dlms_c/src/include/dlms.h:184:24: error: expected primary-expression before 'type'
  184 |         DLMS_DATA_TYPE type,
      |                        ^~~~
~/Arduino/libraries/gx_dlms_c/src/include/dlms.h:185:9: error: 'dlmsVARIANT' was not declared in this scope
  185 |         dlmsVARIANT* newValue);
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/dlms.h:185:22: error: 'newValue' was not declared in this scope
  185 |         dlmsVARIANT* newValue);
      |                      ^~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/dlms.h:185:30: error: expression list treated as compound expression in initializer [-fpermissive]
  185 |         dlmsVARIANT* newValue);
      |                              ^
~/Arduino/libraries/gx_dlms_c/src/include/dlms.h:190:9: error: 'dlmsVARIANT' has not been declared
  190 |         dlmsVARIANT* newValue);
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/client.h:231:9: error: 'dlmsVARIANT' has not been declared
  231 |         dlmsVARIANT* data,
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/client.h:239:9: error: 'dlmsVARIANT' has not been declared
  239 |         dlmsVARIANT* data,
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/client.h:245:9: error: 'dlmsVARIANT' has not been declared
  245 |         dlmsVARIANT* newValue);
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/client.h:251:9: error: 'dlmsVARIANT' has not been declared
  251 |         dlmsVARIANT* value);
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/client.h:311:9: error: 'dlmsVARIANT' has not been declared
  311 |         dlmsVARIANT* data,
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/client.h:355:9: error: 'dlmsVARIANT' has not been declared
  355 |         dlmsVARIANT* data,
      |         ^~~~~~~~~~~
In file included from ~/Arduino/libraries/gx_dlms_c/src/gx_dlms_c.h:21:
~/Arduino/libraries/gx_dlms_c/src/include/cosem.h:159:42: error: 'dlmsVARIANT' has not been declared
  159 |   int cosem_getVariant(gxByteBuffer* bb, dlmsVARIANT* value);
      |                                          ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/cosem.h:163:47: error: 'dlmsVARIANT' has not been declared
  163 |   int cosem_getVariantExact(gxByteBuffer* bb, dlmsVARIANT* value);
      |                                               ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/cosem.h:207:42: error: 'dlmsVARIANT' has not been declared
  207 |   int cosem_setVariant(gxByteBuffer* bb, dlmsVARIANT* value);
      |                                          ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/cosem.h:232:7: error: 'dlmsVARIANT' has not been declared
  232 |       dlmsVARIANT* parameters,
      |       ^~~~~~~~~~~
In file included from ~/Arduino/libraries/gx_dlms_c/src/include/gxset.h:43,
                 from ~/Arduino/libraries/gx_dlms_c/src/gx_dlms_c.h:46:
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:51:68: error: 'dlmsVARIANT' has not been declared
   51 |     int cosem_setRegister(gxRegister* object, unsigned char index, dlmsVARIANT* value);
      |                                                                    ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:55:86: error: 'dlmsVARIANT' has not been declared
   55 |     int cosem_setClock(dlmsSettings* settings, gxClock* object, unsigned char index, dlmsVARIANT* value);
      |                                                                                      ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:117:104: error: 'dlmsVARIANT' has not been declared
  117 |     int cosem_setProfileGeneric(dlmsSettings* settings, gxProfileGeneric* object, unsigned char index, dlmsVARIANT* value);
      |                                                                                                        ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:176:80: error: 'dlmsVARIANT' has not been declared
  176 |     int cosem_setMbusDiagnostic(gxMbusDiagnostic* object, unsigned char index, dlmsVARIANT* value);
      |                                                                                ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:179:78: error: 'dlmsVARIANT' has not been declared
  179 |     int cosem_setMbusPortSetup(gxMBusPortSetup* object, unsigned char index, dlmsVARIANT* value);
      |                                                                              ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:186:94: error: 'dlmsVARIANT' has not been declared
  186 |     int cosem_setG3PlcMacLayerCounters(gxG3PlcMacLayerCounters* object, unsigned char index, dlmsVARIANT* value);
      |                                                                                              ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:189:78: error: 'dlmsVARIANT' has not been declared
  189 |     int cosem_setG3PlcMacSetup(gxG3PlcMacSetup* object, unsigned char index, dlmsVARIANT* value);
      |                                                                              ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:192:76: error: 'dlmsVARIANT' has not been declared
  192 |     int cosem_setG3Plc6LoWPAN(gxG3Plc6LoWPAN* object, unsigned char index, dlmsVARIANT* value);
      |                                                                            ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:199:9: error: 'dlmsVARIANT' has not been declared
  199 |         dlmsVARIANT* value);
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:202:100: error: 'dlmsVARIANT' has not been declared
  202 |     int cosem_setArrayManager(dlmsSettings* settings, gxArrayManager* object, unsigned char index, dlmsVARIANT* value);
      |                                                                                                    ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:332:9: error: 'dlmsVARIANT' has not been declared
  332 |         dlmsVARIANT* value);
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:340:9: error: 'dlmsVARIANT' has not been declared
  340 |         dlmsVARIANT* value);
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:348:9: error: 'dlmsVARIANT' has not been declared
  348 |         dlmsVARIANT* value);
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:356:9: error: 'dlmsVARIANT' has not been declared
  356 |         dlmsVARIANT* value);
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:364:9: error: 'dlmsVARIANT' has not been declared
  364 |         dlmsVARIANT* value);
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:372:9: error: 'dlmsVARIANT' has not been declared
  372 |         dlmsVARIANT* value);
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:380:9: error: 'dlmsVARIANT' has not been declared
  380 |         dlmsVARIANT* value);
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:388:9: error: 'dlmsVARIANT' has not been declared
  388 |         dlmsVARIANT* value);
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:396:9: error: 'dlmsVARIANT' has not been declared
  396 |         dlmsVARIANT* value);
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:400:68: error: 'dlmsVARIANT' has not been declared
  400 |     int cosem_setNtpSetup(gxNtpSetup* object, unsigned char index, dlmsVARIANT* value);
      |                                                                    ^~~~~~~~~~~
~/XLabs/Projects/gurux_arduino/GXDLMSClient.h:74:19: error: 'dlmsVARIANT' has not been declared
   74 |                   dlmsVARIANT *value);
      |                   ^~~~~~~~~~~
~/XLabs/Projects/gurux_arduino/GXDLMSClient.h:114:53: error: 'dlmsVARIANT' has not been declared
  114 |   int Method(gxObject *object, unsigned char index, dlmsVARIANT *data,
      |                                                     ^~~~~~~~~~~
~/XLabs/Projects/gurux_arduino/GXDLMSClient.cpp:140:81: error: 'dlmsVARIANT' has not been declared
  140 | int GXDLMSClient::UpdateValue(gxObject* object, unsigned char attributeOrdinal, dlmsVARIANT* value)
      |                                                                                 ^~~~~~~~~~~
~/XLabs/Projects/gurux_arduino/GXDLMSClient.cpp:165:65: error: 'dlmsVARIANT' has not been declared
  165 | int GXDLMSClient::Method(gxObject* object, unsigned char index, dlmsVARIANT* data, message* messages)
      |                                                                 ^~~~~~~~~~~
~/XLabs/Projects/gurux_arduino/GXDLMSClient.cpp:384:2: error: expected '}' at end of input
  384 | }
      |  ^
~/Arduino/libraries/gx_dlms_c/src/gx_dlms_c.h:5:12: note: to match this '{'
    5 | extern "C" {
      |            ^
In file included from ~/Arduino/libraries/gx_dlms_c/src/include/objectarray.h:41,
                 from ~/Arduino/libraries/gx_dlms_c/src/include/dlmssettings.h:42,
                 from ~/Arduino/libraries/gx_dlms_c/src/include/apdu.h:36,
                 from ~/Arduino/libraries/gx_dlms_c/src/gx_dlms_c.h:12,
                 from ~/XLabs/Projects/gurux_arduino/GXDLMSClient.h:39,
                 from ~/XLabs/Projects/gurux_arduino/gurux_arduino.ino:37:
~/Arduino/libraries/gx_dlms_c/src/include/gxobjects.h:430:9: error: 'dlmsVARIANT' does not name a type
  430 |         dlmsVARIANT value;
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/gxobjects.h:444:9: error: 'dlmsVARIANT' does not name a type
  444 |         dlmsVARIANT value;
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/gxobjects.h:3426:9: error: 'bitArray' does not name a type
 3426 |         bitArray toneMap;
      |         ^~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/gxobjects.h:3438:9: error: 'bitArray' does not name a type
 3438 |         bitArray txCoeff;
      |         ^~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/gxobjects.h:3484:9: error: 'bitArray' does not name a type
 3484 |         bitArray toneMask;
      |         ^~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/gxobjects.h:3715:9: error: 'variantArray' does not name a type
 3715 |         variantArray groupTable;
      |         ^~~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/gxobjects.h:3765:9: error: 'variantArray' does not name a type
 3765 |         variantArray destinationAddress;
      |         ^~~~~~~~~~~~
In file included from ~/Arduino/libraries/gx_dlms_c/src/include/dlmssettings.h:44:
~/Arduino/libraries/gx_dlms_c/src/include/gxvalueeventargs.h:48:9: error: 'dlmsVARIANT' does not name a type
   48 |         dlmsVARIANT value;
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/gxvalueeventargs.h:79:9: error: 'dlmsVARIANT' does not name a type
   79 |         dlmsVARIANT parameters;
      |         ^~~~~~~~~~~
In file included from ~/Arduino/libraries/gx_dlms_c/src/include/dlmssettings.h:45:
~/Arduino/libraries/gx_dlms_c/src/include/replydata.h:72:9: error: 'dlmsVARIANT' does not name a type
   72 |         dlmsVARIANT dataValue;
      |         ^~~~~~~~~~~
In file included from ~/Arduino/libraries/gx_dlms_c/src/include/client.h:46,
                 from ~/Arduino/libraries/gx_dlms_c/src/gx_dlms_c.h:19:
~/Arduino/libraries/gx_dlms_c/src/include/dlms.h:104:9: error: 'dlmsVARIANT' has not been declared
  104 |         dlmsVARIANT* value);
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/dlms.h:119:9: error: 'dlmsVARIANT' has not been declared
  119 |         dlmsVARIANT* value);
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/dlms.h:143:9: error: 'dlmsVARIANT' has not been declared
  143 |         dlmsVARIANT* value);
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/dlms.h:183:9: error: 'dlmsVARIANT' was not declared in this scope
  183 |         dlmsVARIANT* value,
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/dlms.h:183:22: error: 'value' was not declared in this scope
  183 |         dlmsVARIANT* value,
      |                      ^~~~~
~/Arduino/libraries/gx_dlms_c/src/include/dlms.h:184:24: error: expected primary-expression before 'type'
  184 |         DLMS_DATA_TYPE type,
      |                        ^~~~
~/Arduino/libraries/gx_dlms_c/src/include/dlms.h:185:9: error: 'dlmsVARIANT' was not declared in this scope
  185 |         dlmsVARIANT* newValue);
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/dlms.h:185:22: error: 'newValue' was not declared in this scope
  185 |         dlmsVARIANT* newValue);
      |                      ^~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/dlms.h:185:30: error: expression list treated as compound expression in initializer [-fpermissive]
  185 |         dlmsVARIANT* newValue);
      |                              ^
~/Arduino/libraries/gx_dlms_c/src/include/dlms.h:190:9: error: 'dlmsVARIANT' has not been declared
  190 |         dlmsVARIANT* newValue);
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/client.h:231:9: error: 'dlmsVARIANT' has not been declared
  231 |         dlmsVARIANT* data,
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/client.h:239:9: error: 'dlmsVARIANT' has not been declared
  239 |         dlmsVARIANT* data,
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/client.h:245:9: error: 'dlmsVARIANT' has not been declared
  245 |         dlmsVARIANT* newValue);
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/client.h:251:9: error: 'dlmsVARIANT' has not been declared
  251 |         dlmsVARIANT* value);
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/client.h:311:9: error: 'dlmsVARIANT' has not been declared
  311 |         dlmsVARIANT* data,
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/client.h:355:9: error: 'dlmsVARIANT' has not been declared
  355 |         dlmsVARIANT* data,
      |         ^~~~~~~~~~~
In file included from ~/Arduino/libraries/gx_dlms_c/src/gx_dlms_c.h:21:
~/Arduino/libraries/gx_dlms_c/src/include/cosem.h:159:42: error: 'dlmsVARIANT' has not been declared
  159 |   int cosem_getVariant(gxByteBuffer* bb, dlmsVARIANT* value);
      |                                          ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/cosem.h:163:47: error: 'dlmsVARIANT' has not been declared
  163 |   int cosem_getVariantExact(gxByteBuffer* bb, dlmsVARIANT* value);
      |                                               ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/cosem.h:207:42: error: 'dlmsVARIANT' has not been declared
  207 |   int cosem_setVariant(gxByteBuffer* bb, dlmsVARIANT* value);
      |                                          ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/cosem.h:232:7: error: 'dlmsVARIANT' has not been declared
  232 |       dlmsVARIANT* parameters,
      |       ^~~~~~~~~~~
In file included from ~/Arduino/libraries/gx_dlms_c/src/include/gxset.h:43,
                 from ~/Arduino/libraries/gx_dlms_c/src/gx_dlms_c.h:46:
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:51:68: error: 'dlmsVARIANT' has not been declared
   51 |     int cosem_setRegister(gxRegister* object, unsigned char index, dlmsVARIANT* value);
      |                                                                    ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:55:86: error: 'dlmsVARIANT' has not been declared
   55 |     int cosem_setClock(dlmsSettings* settings, gxClock* object, unsigned char index, dlmsVARIANT* value);
      |                                                                                      ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:117:104: error: 'dlmsVARIANT' has not been declared
  117 |     int cosem_setProfileGeneric(dlmsSettings* settings, gxProfileGeneric* object, unsigned char index, dlmsVARIANT* value);
      |                                                                                                        ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:176:80: error: 'dlmsVARIANT' has not been declared
  176 |     int cosem_setMbusDiagnostic(gxMbusDiagnostic* object, unsigned char index, dlmsVARIANT* value);
      |                                                                                ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:179:78: error: 'dlmsVARIANT' has not been declared
  179 |     int cosem_setMbusPortSetup(gxMBusPortSetup* object, unsigned char index, dlmsVARIANT* value);
      |                                                                              ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:186:94: error: 'dlmsVARIANT' has not been declared
  186 |     int cosem_setG3PlcMacLayerCounters(gxG3PlcMacLayerCounters* object, unsigned char index, dlmsVARIANT* value);
      |                                                                                              ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:189:78: error: 'dlmsVARIANT' has not been declared
  189 |     int cosem_setG3PlcMacSetup(gxG3PlcMacSetup* object, unsigned char index, dlmsVARIANT* value);
      |                                                                              ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:192:76: error: 'dlmsVARIANT' has not been declared
  192 |     int cosem_setG3Plc6LoWPAN(gxG3Plc6LoWPAN* object, unsigned char index, dlmsVARIANT* value);
      |                                                                            ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:199:9: error: 'dlmsVARIANT' has not been declared
  199 |         dlmsVARIANT* value);
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:202:100: error: 'dlmsVARIANT' has not been declared
  202 |     int cosem_setArrayManager(dlmsSettings* settings, gxArrayManager* object, unsigned char index, dlmsVARIANT* value);
      |                                                                                                    ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:332:9: error: 'dlmsVARIANT' has not been declared
  332 |         dlmsVARIANT* value);
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:340:9: error: 'dlmsVARIANT' has not been declared
  340 |         dlmsVARIANT* value);
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:348:9: error: 'dlmsVARIANT' has not been declared
  348 |         dlmsVARIANT* value);
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:356:9: error: 'dlmsVARIANT' has not been declared
  356 |         dlmsVARIANT* value);
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:364:9: error: 'dlmsVARIANT' has not been declared
  364 |         dlmsVARIANT* value);
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:372:9: error: 'dlmsVARIANT' has not been declared
  372 |         dlmsVARIANT* value);
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:380:9: error: 'dlmsVARIANT' has not been declared
  380 |         dlmsVARIANT* value);
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:388:9: error: 'dlmsVARIANT' has not been declared
  388 |         dlmsVARIANT* value);
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:396:9: error: 'dlmsVARIANT' has not been declared
  396 |         dlmsVARIANT* value);
      |         ^~~~~~~~~~~
~/Arduino/libraries/gx_dlms_c/src/include/../include/gxsetmalloc.h:400:68: error: 'dlmsVARIANT' has not been declared
  400 |     int cosem_setNtpSetup(gxNtpSetup* object, unsigned char index, dlmsVARIANT* value);
      |                                                                    ^~~~~~~~~~~
~/XLabs/Projects/gurux_arduino/GXDLMSClient.h:74:19: error: 'dlmsVARIANT' has not been declared
   74 |                   dlmsVARIANT *value);
      |                   ^~~~~~~~~~~
~/XLabs/Projects/gurux_arduino/GXDLMSClient.h:114:53: error: 'dlmsVARIANT' has not been declared
  114 |   int Method(gxObject *object, unsigned char index, dlmsVARIANT *data,
      |                                                     ^~~~~~~~~~~
~/XLabs/Projects/gurux_arduino/gurux_arduino.ino:509:66: error: 'dlmsVARIANT' has not been declared
  509 | int com_method(gxObject *object, unsigned char attributeOrdinal,
      |                                                                  ^          
~/XLabs/Projects/gurux_arduino/gurux_arduino.ino: In function 'int com_updateInvocationCounter(const char*)':
~/XLabs/Projects/gurux_arduino/gurux_arduino.ino:284:38: error: 'struct gxData' has no member named 'value'
  284 |                     var_toInteger(&d.value));
      |                                      ^~~~~
~/XLabs/Projects/gurux_arduino/gurux_arduino.ino:284:21: error: 'var_toInteger' was not declared in this scope; did you mean 'ba_toInteger'?
  284 |                     var_toInteger(&d.value));
      |                     ^~~~~~~~~~~~~
      |                     ba_toInteger
~/XLabs/Projects/gurux_arduino/gurux_arduino.ino:285:35: error: 'struct gxData' has no member named 'value'
  285 |         ic = 1 + var_toInteger(&d.value);
      |                                   ^~~~~
~/XLabs/Projects/gurux_arduino/gurux_arduino.ino: In function 'int com_read(gxObject*, unsigned char)':
~/XLabs/Projects/gurux_arduino/gurux_arduino.ino:485:66: error: 'struct gxReplyData' has no member named 'dataValue'
  485 |       (ret = Client.UpdateValue(object, attributeOrdinal, &reply.dataValue)) !=
      |                                                                  ^~~~~~~~~
~/XLabs/Projects/gurux_arduino/gurux_arduino.ino: At global scope:
~/XLabs/Projects/gurux_arduino/gurux_arduino.ino:510:16: error: 'dlmsVARIANT' has not been declared
  510 |                dlmsVARIANT *params) {
      |                ^~~~~~~~~~~
~/XLabs/Projects/gurux_arduino/gurux_arduino.ino: In function 'int com_readRowsByEntry(gxProfileGeneric*, long unsigned int, long unsigned int)':
~/XLabs/Projects/gurux_arduino/gurux_arduino.ino:579:63: error: 'struct gxReplyData' has no member named 'dataValue'
  579 |       (ret = Client.UpdateValue((gxObject *)object, 2, &reply.dataValue)) !=
      |                                                               ^~~~~~~~~
~/XLabs/Projects/gurux_arduino/gurux_arduino.ino: In function 'int com_readRowsByRange(gxProfileGeneric*, gxtime*, gxtime*)':
~/XLabs/Projects/gurux_arduino/gurux_arduino.ino:597:63: error: 'struct gxReplyData' has no member named 'dataValue'
  597 |       (ret = Client.UpdateValue((gxObject *)object, 2, &reply.dataValue)) !=
      |                                                               ^~~~~~~~~
~/XLabs/Projects/gurux_arduino/gurux_arduino.ino: In function 'int com_readAllObjects(const char*)':
~/XLabs/Projects/gurux_arduino/gurux_arduino.ino:937:5: error: 'obj_toString' was not declared in this scope; did you mean 'ba_toString'?
  937 |     obj_toString(BASE(ldn), &data);
      |     ^~~~~~~~~~~~
      |     ba_toString
~/XLabs/Projects/gurux_arduino/gurux_arduino.ino:948:5: error: 'obj_toString' was not declared in this scope; did you mean 'ba_toString'?
  948 |     obj_toString(BASE(clock1), &data);
      |     ^~~~~~~~~~~~
      |     ba_toString
~/XLabs/Projects/gurux_arduino/gurux_arduino.ino: At global scope:
~/XLabs/Projects/gurux_arduino/gurux_arduino.ino:1023:2: error: expected '}' at end of input
 1023 | }
      |  ^
~/Arduino/libraries/gx_dlms_c/src/gx_dlms_c.h:5:12: note: to match this '{'
    5 | extern "C" {
      |            ^

Error during build: exit status 1
IntelliSense configuration already up to date. To manually rebuild your IntelliSense configuration run "Ctrl+Alt+I"
[Error] Verifying sketch 'gurux_arduino.ino': Exit with code=1

```

## RCA

- Changing inclusion order in `gx_dlms_c` didn't have any effect on the errors. Concluded that it's not inclusion order.
- After inspection, it was clear that the file `variant.h` wasn't being included. Tried to remove the header guard from it, the linker threw multiple-definition errors.
- Changing the header guard value fixed the errors.
- further inspection yielded that STM uses the deefinition `VARIANT_H`, as a result, the file `variant.h` was included as an empty file.

## Proposed Solution

- Change the header guard to anything other than `VARIANT_H`

--- a/src/include/variant.h
+++ b/src/include/variant.h
@@ -30,8 +30,8 @@
 // Full text may be retrieved at http://www.gnu.org/licenses/gpl-2.0.txt
 //---------------------------------------------------------------------------
 
-#ifndef VARIANT_H
-#define VARIANT_H
+#ifndef GX_VARIANT_H
+#define GX_VARIANT_H
 
 #ifdef  __cplusplus
 extern "C" {
@@ -608,4 +608,4 @@ extern "C" {
 }
 #endif
 
-#endif //VARIANT_H
+#endif //GX_VARIANT_H

